sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java

```diff
@@ -2521,10 +2508,280 @@ public void testDetermineCharsetFromMessageASCIISpecialCase(@TempDir final File
+                    messageStorage);
+            final String content = String.format(
+                    "<?xml version=\"1.0\" encoding=\"%s\"?>%n<sometag>üöä</sometag>", charsetInXMLDeclaration);
+            final byte[] encodedContent = content.getBytes(StandardCharsets.UTF_8);
+            message.write(encodedContent, 0, encodedContent.length);
+            message.close();
+
+            // when
+            final var result = messageStorage.convertMessageToMessageContent(message);
+            // then
+            assertNotNull(result);
+        }
+        verify(testRunObserver, times(4)).invalidateTestRun(anyString());
+    }
+
+    /**
+     * Checks that convertMessageToMessageContent() works when it does not detect any decoding problems in the message.
+     * @param dir - a temporary directory.
+     * @throws IOException - when something goes wrong.
+     */
+    @Test
+    public void testConvertToMessageContentFailOnEncodingProblem2(@TempDir final File dir) throws IOException {
+
+        final Charset charsetInHttpHeader = StandardCharsets.UTF_8;
+        final Charset charsetInXMLDeclaration = StandardCharsets.UTF_8;
+        final String mimeType = "application/xml";
+
+        try (final MessageStorage messageStorage = new MessageStorage(
+                1, false, true, mock(MessageFactory.class), new HibernateConfigImpl(dir), this.testRunObserver)) {
+            // given
+            final ListMultimap<String, String> headers = ArrayListMultimap.create();
+            headers.put("Content-Type", String.format("%s;charset=%s", mimeType, charsetInHttpHeader));
+
+            final HttpApplicationInfo applicationInfo = new HttpApplicationInfo(headers, "transactionId", "requestURI");
+            final TransportInfo transportInfo =
+                    new TransportInfo("http", "localhost", 1234, "remotehost", 4567, List.of());
+            final CommunicationContext communicationContext = new CommunicationContext(applicationInfo, transportInfo);
+            final Message message = new Message(
+                    CommunicationLog.Direction.INBOUND,
+                    CommunicationLog.MessageType.REQUEST,
+                    communicationContext,
+                    messageStorage);
+            final String content = String.format(
+                    "<?xml version=\"1.0\" encoding=\"%s\"?>%n<sometag>üöä</sometag>", charsetInXMLDeclaration);
+            final byte[] encodedContent = content.getBytes(StandardCharsets.UTF_16);
+
+            message.write(encodedContent, 0, encodedContent.length);
+            message.close();
+
+            // when
+            final var result = messageStorage.convertMessageToMessageContent(message);
+
+            // then
+            assertNotNull(result);
+        }
+        verify(testRunObserver, times(6)).invalidateTestRun(anyString());
+    }
+
+    /**
+     * Checks that convertMessageToMessageContent() works when it does not detect any decoding problems in the message.
```

Comment:
this is equal to the description for testConvertToMessageContent

has to be changed, since it does not fit here

Reply:
Done.